### PR TITLE
This removes support for dynamic API backend lookup via DNS SRV records

### PR DESCRIPTION
**Pull request by arlandism**
Per issue #2418, the DNS SRV record functionality was a headache to maintain because of security concerns and become moot since everything is fetched from rubygems.org.

# Description:
- Remove support for dynamic API backend lookup via DNS SRV records
- Remove related code in RemoteFetcher and Source
- Update broken link in CONTRIBUTING.md
______________

# Tasks:

- [x] Describe the problem / feature
- [x] Remove tests
- [x] Remove code to solve the problem
- [x] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

I went ahead and aggressively removed parts that seemed unnecessary now that the functionality is being removed. Let me know if I was *too* aggressive and should pare back a bit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ See http://bundler.io/compatibility for known issues.
 
 ### Contributing
 
-If you'd like to contribute to RubyGems, that's awesome, and we <3 you. Check out our [guide to contributing](https://github.com/rubygems/rubygems/blob/master/CONTRIBUTING.rdoc#how-to-contribute) for more information.
+If you'd like to contribute to RubyGems, that's awesome, and we <3 you. Check out our [guide to contributing](CONTRIBUTING.md) for more information.
```
Comment (bronzdoc) [inline README.md]: This is not related to `Removing support for dynamic API backend` maybe in another PR? :)

Comment (arlandism) [reply README.md]: Sure thing

**Comment by arlandism** [comment]
I'll be looking into the Travis failures today

**Comment by arlandism** [comment]
hey @bronzdoc, is there a good way to replicate these failures from a local environment? I've tried setting `RGV` to my branch and running the specs from bundler via `bin/rake spec:rubygems:master`, but everything still passes.

**Comment by refashioned** [comment]
I guess the Travis failures are because [`Bundler::GemRemoteFetcher` inherits from `Gem::RemoteFetcher`](https://github.com/bundler/bundler/blob/970ac8978a79fc5da15cd969baeab3133bc5cf31/lib/bundler/gem_remote_fetcher.rb#L11) and these changes [modify the signature of `Gem::RemoteFetcher`](https://github.com/rubygems/rubygems/pull/2433/files#diff-8f942ed54b046e4976592502746d9c6fR77). E.g., [here's](https://github.com/bundler/bundler/blob/970ac8978a79fc5da15cd969baeab3133bc5cf31/lib/bundler/rubygems_integration.rb#L779-L780) the code passing a `Resolv::DNS` into `Bundler::GemRemoteFetcher`.

[In `pretty_print`](https://github.com/rubygems/rubygems/pull/2433/files#diff-2d4d043362c2fdd14a9ee8973f27c7b8L223), I wonder if you can remove the entire `'API URI: '` conditional block? Seeing as there's no separate API URI anymore. I manually examined all the places where you changed `api_uri` to `uri`, and this is the only one that stood out.

**Comment by arlandism** [comment]
@refashioned Good call on both counts!

I managed to get this working by adding an unused, default `dns` parameter in `Gem::RemoteFetcher.initialize`,  which I'm not a huge fan of but gets around breaking the function signature. 

Thoughts?

**Comment by jrochkind** [comment]
> I managed to get this working by adding an unused, default dns parameter in 

Should maybe doc in source comments what's going on here, that this arg is for backwards compat but is intentionally unused? Otherwise it's a bit confusing if someone comes across it in source code, a developer's default assumption is that an arg is used by something, or it wouldn't be there. 

Apologies if I missed comment docs that are already there!
